packages/base/source/adt/union/union.js

```diff
@@ -82,8 +108,11 @@ instead to check if a value belongs to the ADT variant.`);
        *   where 'b = 'a[`@@folktale:adt:tag]
        */
       matchWith(pattern) {
-        return pattern[name](this);
-      } 
+        assertObject('Union.mapObject#matchWith', pattern);
+        if (name in pattern) { return pattern[name](this); }
+        else if (ANY in pattern) { return pattern[ANY](this) }
```

Comment:
... actually, sorry, coffee kicking in, that was point; like you don't care about the arguments, they're supposed to be ignored. Fixing test...